src/graphics/program-lib/programs/standard-node.js

```diff
@@ -0,0 +1,93 @@
+import { standard } from './standard.js';
+
+// just do a shallow copy standard
+var standardnode = Object.assign({}, standard);
+
+// override a couple of functions - the overridden functions have been altered in standard to accept extra params
+standardnode._generateKey = standard.generateKey;
+standardnode.generateKey = function (options) {
+    return this._generateKey(options, options._shaderGraphChunk.id);
+};
+
+standardnode._createShaderDefinition = standard.createShaderDefinition;
+standardnode.createShaderDefinition = function (device, options) {
+    var rootShaderGraph = null;
+    var rootDeclGLSL = '';
+    var rootCallGLSL = '';
+
+    if (options._shaderGraphChunk) {
+        rootShaderGraph = options._shaderGraphChunk;
+
+        rootDeclGLSL = rootShaderGraph.generateRootDeclGlsl();
+        rootCallGLSL = rootShaderGraph.generateRootCallGlsl();
+    }
+
+    var graphCode = [];
+
+    if (options._shaderGraphChunk) {
```

Comment:
any reason this condition is repeated few times below, instead of all code being inside a single if statement?

Reply:
Its just a hangover from actually being embedded in the standard material shader generator - where it was useful for ironing out edge cases in the logic - I will remove the duplication - as put comments in to help maintenance when StandardMaterial is updated

Reply:
I have improved this - I haven't turned it into a generic data driven system because it is still tied to the non-data driven standard material.